WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,166 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using System.Threading;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		public bool RequestEnd { get; set; }
+		public Network Network { get; }
+		public Task? ScanningTask { get; set; }
+
+		public WebcamQrReader(Network network)
+		{
+			Network = network;
+		}
+
+		public void StartScanning()
+		{
+			if (ScanningTask is not { })
+			{
+				ScanningTask = Task.Run(() =>
+				{
+					VideoCapture? camera = null;
+					try
+					{
+						if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+						{
+							throw new NotImplementedException("This operating system is not supported.");
+						}
+						camera = OpenCamera();
+						RequestEnd = false;
+						Scan(camera);
+					}
+					catch (Exception exc)
+					{
+						Logger.LogError("QR scanning stopped. Reason:", exc);
+						ErrorOccured?.Invoke(this, exc);
+					}
+					finally
+					{
+						camera?.Release();
+					}
+				});
+			}
+		}
+
+		public async Task StopScanningAsync()
+		{
+			if (ScanningTask is { } task)
+			{
+				RequestEnd = true;
+				await task;
+
+				ScanningTask = null;
+			}
+		}
+
+		private void Scan(VideoCapture camera)
+		{
+			WriteableBitmap? lastBitmap = null;
+			WriteableBitmap? currentBitmap = null;
+			using QRCodeDetector qRCodeDetector = new();
+			while (!RequestEnd)
+			{
+				try
+				{
+					using Mat frame = new();
+					camera.Read(frame);
```

Comment:
The docs states that this returns `false` if no frame has been grabbed. Should the return value be handled?

Reply:
In that case `frame.Empty()` check should do the same work, question is which way is nicer use?

Reply:
I would rather check return value of `camera.Read(frame)` to avoid the question of how costly `frame.Empty()` is. It may be a very cheap call or it may be a costly one.